### PR TITLE
Update to Kotlin 1.9.0

### PR DESCRIPTION
**Pull request by ZacSweers**

### CODE REVIEW

**Comment by ZacSweers** [comment]
I've released a `0.22.1-1.9.0-Beta` with the changes up to this point in the PR

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -55,7 +55,7 @@ incap = { module = "net.ltgt.gradle.incap:incap", version.ref = "incap" }
 incap-processor = { module = "net.ltgt.gradle.incap:incap-processor", version.ref = "incap" }
 
 kotlin-compilerEmbeddable = { module = "org.jetbrains.kotlin:kotlin-compiler-embeddable", version.ref = "kotlin" }
-kotlin-metadata = { module = "org.jetbrains.kotlinx:kotlinx-metadata-jvm", version = "0.6.0" }
+kotlin-metadata = { module = "org.jetbrains.kotlinx:kotlinx-metadata-jvm", version = "0.6.1" }
```
Comment (ZacSweers) [inline gradle/libs.versions.toml:58]:
```suggestion
kotlin-metadata = { module = "org.jetbrains.kotlinx:kotlinx-metadata-jvm", version = "0.6.2" }
```

**Comment by sandwwraith** [comment]
NB: binary-compatibility-validator 0.13.2 is out with dependency on metadata 0.6.2

**Comment by ZacSweers** [comment]
Thanks!

**Pull request merged**